src/Compiler/AST/Visitor/ExprConverter.cpp

```diff
@@ -262,55 +266,111 @@ void ExprConverter::ConvertExprImageAccessArray(ExprPtr& expr, ArrayExpr* arrayE
             {
                 /* Get buffer type denoter from array indices of array access plus identifier */
                 auto bufferTypeDen = bufferDecl->GetTypeDenoter()->GetSub(arrayExpr);
-
-                if (!arrayExpr->arrayIndices.empty())
+                if (auto baseBufferTypeDen = bufferTypeDen->As<BaseTypeDenoter>())
```

Comment:
Ok, I'll add this to the tests as `ImageBufferTest1.hlsl`.

`GetSub` derives the type denoter by its input expression.
E.g. `BaseTypeDenoter::GetSub(ObjectExpr)` tries to derive the type for a vector suffix like this:
`float4 vec; vec.xy;` where `xy` is the identifier of the `ObjectExpr` and `vec` is from type `BaseTypeDenoter` with `dataType = Float4`, so `GetSub` will return `BaseTypeDentoer` with `dataType = Float2`, since `xy` is a 2-component-vector.

I know this is something one has to get used to, but the type system is a complex part anyways. That's why I refactored it a while ago.

The test [RWTextureTest1.hlsl](https://github.com/LukasBanana/XShaderCompiler/blob/master/test/RWTextureTest1.hlsl) currently only compiles the first `imageLoad` output correctly, becasue in the second one, the array indices are interpreted as a continuous array access, although the two array indices mean something completely differnt:
```hlsl
RWTexture2D<int> tex[2];
void main() {
    float4 a = (tex[0])[int2(1, 2)]; // must be "imageLoad(tex[0], ivec2(1, 2))" in GLSL
    float4 b =  tex[0] [int2(3, 4)]; // must be "imageLoad(tex[0], ivec2(1, 2))" in GLSL
}
```